File: app/src/main/java/com/example/bookapp/interfaces/FriendsRepositoryInterface.java
package com.example.bookapp.interfaces;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface FriendsRepositoryInterface {

    @GET("RestfulRequestHandler.php")
    Call<String> getFriends(@Query("userID") String userID, @Query("friends") boolean friends, @Query("lastMessage") boolean lastMessage);


}

File: app/src/main/java/com/example/bookapp/interfaces/FriendsRepositoryInterface.kt
package com.example.bookapp.interfaces

import com.example.dataLayer.models.FriendDTO
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Query
import java.util.*

interface FriendsRepositoryInterface {
    @GET("RestfulRequestHandler.php")
    fun getFriends(@Query("userID") userID: String?, @Query("friends") friends: Boolean, @Query("lastMessage") lastMessage: Boolean): Call<ArrayList<FriendDTO>>
}